### PR TITLE
Determining fibonacci number

### PR DESCRIPTION
**Pull request by paraganz**
Fixes #   https://github.com/anz-bank/go-training/issues/24

Changes proposed in this PR:
- Printing Fibonacci series

### CODE REVIEW

File: 00_hello_world/main.go [hunk 1]
```diff
@@ -8,6 +8,21 @@ import (
 
 var out io.Writer = os.Stdout
 
+func fib(i int) int {
+	var a = 0
+	var b = 1
```
Comment (anzdaddy) [inline 00_hello_world/main.go]: These two lines could be written `a, b := 0, 1`.

File: 00_hello_world/main.go [hunk 2]
```diff
@@ -8,6 +8,21 @@ import (
...
+	fmt.Fprintln(out, b)
+	var sum int
```
Comment (anzdaddy) [inline 00_hello_world/main.go]: Although `sum` happens to default to zero, it's better to express the intent that it must start from zero: `sum := 0`.

File: 00_hello_world/main.go [hunk 3]
```diff
@@ -8,6 +8,21 @@ import (
...
+	for cnt := 0; cnt < i-1; cnt++ {
+		sum = a + b
+		fmt.Fprintln(out, sum)
+		a = b
+		b = sum
```
Comment (anzdaddy) [inline 00_hello_world/main.go]: You can update in one go: `a, b = b, sum`.

File: 00_hello_world/main_test.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ func TestMainOutput(t *testing.T) {
 	main()
 
 	// Then
-	expected := strconv.Quote("Hallo du schöne Welt!\n")
+	expected := strconv.Quote("Hallo du schöne Welt!\n1\n1\n2\n3\n5\n8\n13\n")
```
Comment (anzdaddy) [inline 00_hello_world/main_test.go]: This could be:
```go
	expected := strconv.Quote("Hallo du schöne Welt!
1
1
2
3
5
8
13
`)
```
It's not unequivocally the right thing to do, however. It improves readability in one dimension, but hampers in another by breaking the indentation structure. It's a judgment call, really.

Comment (paraganz) [reply 00_hello_world/main_test.go]: Done. Thanks.

File: 00_hello_world/main.go (hunk 3 above)
Comment (paraganz) [reply 00_hello_world/main.go]: Done. Thanks.

File: 00_hello_world/main.go (hunk 2 above)
Comment (paraganz) [reply 00_hello_world/main.go]: Done. Thanks.

File: 00_hello_world/main.go (hunk 1 above)
Comment (paraganz) [reply 00_hello_world/main.go]: Done. Thanks.

File: 00_hello_world/main.go (hunk 3 above)
Comment (anzdaddy) [reply 00_hello_world/main.go]: It also occurred to me that you don't need sum at all. I'll let you figure it out. 🙂

File: 00_hello_world/main.go [hunk 5]
```diff
@@ -8,6 +8,19 @@ import (
```
Comment (juliaogris) [inline 00_hello_world/main.go]: Move your code from `00_hello_world` to `01_fib`

File: 00_hello_world/main.go [hunk 6]
```diff
@@ -8,6 +8,19 @@ import (
 
 var out io.Writer = os.Stdout
 
+func fib(i int) int {
+	a, b := 0, 1
```
Comment (juliaogris) [inline 00_hello_world/main.go]: This can be simplified to:
```go
	a, b := 1, 1
	for i := 1; i <= n; i++ {
		fmt.Fprintln(out, a)
		a, b = b, a+b
	}
```

File: 00_hello_world/main_test.go [hunk 7]
```diff
@@ -18,7 +18,15 @@ func TestMainOutput(t *testing.T) {
 	main()
 
 	// Then
-	expected := strconv.Quote("Hallo du schöne Welt!\n")
```
Comment (juliaogris) [inline 00_hello_world/main_test.go]: Also test `fib`, not just `main` (unit tests).
Test corner cases - i: -1, -0, 1, very large number

**Review by juliaogris — COMMENTED** [review]
Please add PR description according to [template](https://github.com/anz-bank/go-training/blob/master/.github/pull_request_template.md).

Please also fix up commit messages with `git rebase -i HASH` according to [recommendations](https://chris.beams.io/posts/git-commit/):
* Capitalize the subject line
* Use the imperative mood in the subject line
* Fold fixup commits into previous ones with `git rebase -i`

File: 00_hello_world/main.go [hunk 8]
```diff
@@ -8,6 +8,19 @@ import (
...
+	fmt.Fprintln(out, b)
```
Comment (anz-rfc) [inline 00_hello_world/main.go]: It's better to decouple fibonacci sequence calculation from IO. https://github.com/anz-bank/go-training/issues/37

File: 00_hello_world/main.go (hunk 5 above)
Comment (paraganz) [reply 00_hello_world/main.go]: Done

File: 01_fib/main.go [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+var out io.Writer = os.Stdout
+
+func fib(i int) {
+	a, b := 1, 1
+	for cnt := 0; cnt < i; cnt++ {
+		fmt.Fprintln(out, a)
+		a, b = b, a+b
+	}
+	return
```
Comment (golangcibot) [inline 01_fib/main.go]: redundant return statement (from `gosimple`)
